src/value.rs

```diff
@@ -27,14 +29,21 @@ pub struct AllocatedValue {
 #[derive(Copy, Clone, Debug)]
 pub struct AllocatedQuantity {
     pub variable: Variable,
-    pub assignment: Option<u64>,
+    pub assignment: Option<SignedInteger>,
+}
+
+/// Represents a signed integer in the range [-(2^64-1) .. 2^64-1]
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum SignedInteger {
+    Positive(u64),
```

Comment:
Probably need a comment that each variant is allowed to hold `0` despite the name.